### ISSUE TITLE
Regression for for >=qt-5.8 Can't find QtPlatformSupport header 

### ISSUE THREAD

**Issue by khurshid-alam** [yunit-io/oxide-qt#1]
Qt5PlatformSupport now breaks into  Qt5FontDatabaseSupport, Qt5ThemeSupport and Qt5EventDispatcherSupport

https://git.reviewboard.kde.org/r/129268/diff/2-3#0

As a result oxide-qt build fails against qt-5.9

```
CMake Error at oxide/build/cmake/FindQt5PlatformSupport.cmake:31 (message):
  Can't find QtPlatformSupport header
Call Stack (most recent call first):
  oxide/qt/CMakeLists.txt:36 (find_package)
```

Qt5PlatformSupport is only required if we enabled tests. But I am not sure how to disable test during compilation (from debain/rules may be ?).

**Comment by jsalatas** [comment]
I have a patch for this. I'll send it later today.....

is ubuntu 17.10 using gcc7? if this is the case then there might be other issues which I haven't resolved yet.

**Comment by khurshid-alam** [comment]
I have a patch for this. I'll send it later today.....
Thanks. 

> is ubuntu 17.10 using gcc7?
Yes. But it also has gcc-6 in the archive, so it can compile against gcc-6 too.

**Comment by jsalatas** [comment]
Try the attached patch 
[0002-Port-to-Qt-5.9.zip](https://github.com/yunit-io/oxide-qt/files/1282622/0002-Port-to-Qt-5.9.zip)

**Comment by khurshid-alam** [comment]
Does it compile for you in debian unstable (with patch)?

**Comment by jsalatas** [comment]
> Does it compile for you in debian unstable (with patch)?

it needs additional patches for compiling with gcc7 and I'm still working on it. I have adapted the following patch

https://aur.archlinux.org/cgit/aur.git/tree/chromium-blink-gcc7.patch?h=chromium-vaapi

and some other stuff but I don't have a correct patch so far: it compiles but most of the unit tests fail.

**Comment by khurshid-alam** [comment]
Yes it seems other unit tests fails against gcc-7.

At the moment I am trying to compile it against gcc-6 + qt-5.9.  But the size is too big when I try to push to launchpad from my local repo. The connection is simply dies out. 
Can you publish those patches (with the rest of the repository) in a separate branch? Once published here, I can directly import from this repo.

Thanks.

**Comment by jsalatas** [comment]
Yeah! Its huge size and the resources required to get it compile (cpu, memory and time) make it a pain in the ass. I'll try to submit what I have so far. I am at a point that I can have it compiled with gcc7 (under debian sid) but almost every test fails. Hope I'll submit these soon. 

In the mean time I'm trying to figure out how it works and see if I can upgrade it to a newer chromium version.

**Comment by jsalatas** [comment]
Sorry for the delay. Seems I'm stack with the oxide-qt and I think we can proceed skipping this for now. See my post at https://forum.yunit.io/viewtopic.php?f=4&t=344&p=732#p732

**Comment by khurshid-alam** [comment]
Thanks. I agree with  Alan Griffiths that we could drop oxide and use qtwebengine instead. No oxide means a big chunk of resources is spared and we can use that resource elsewhere. Only online-accounts and ubuntu-web need it. So we could just port those to use qtwebengine, and move on.
